modules/tensor_mechanics/src/utils/RankTwoScalarTools.C

```diff
@@ -12,10 +12,79 @@
 namespace RankTwoScalarTools
 {
 
+MooseEnum
+scalarOptions()
+{
+  return MooseEnum("VonMisesStress EquivalentPlasticStrain Hydrostatic L2norm MaxPrincipal MidPrincipal MinPrincipal VolumetricStrain FirstInvariant SecondInvariant ThirdInvariant AxialStress HoopStress RadialStress TriaxialityStress Direction");
+}
+
+Real
+getQuantity(const RankTwoTensor & tensor, const MooseEnum scalar_type, const Point & point1, const Point & point2, const Point & curr_point, Point & direction)
+{
+  Real val = 0.0;
+
+  switch (scalar_type)
+  {
+    case 0:
+      val = vonMisesStress(tensor);
+      break;
+    case 1:
+      ///For plastic strain tensor (ep), tr(ep) = 0 is considered
+      val = equivalentPlasticStrain(tensor);
+      break;
+    case 2:
+      val = hydrostatic(tensor);
+      break;
+    case 3:
+      val = L2norm(tensor);
+      break;
+    case 4:
+      val = maxPrinciple(tensor);
+      break;
+    case 5:
+      val = midPrinciple(tensor);
+      break;
+    case 6:
+      val = minPrinciple(tensor);
+      break;
+    case 7:
+      val = volumetricStrain(tensor);
+      break;
+    case 8:
+      val = firstInvariant(tensor);
+      break;
+    case 9:
+      val = secondInvariant(tensor);
+      break;
+    case 10:
+      val = thirdInvariant(tensor);
+      break;
+    case 11:
+      val = axialStress(tensor, point1, point2, direction);
+      break;
+    case 12:
+      val = hoopStress(tensor, point1, point2, curr_point, direction);
+      break;
+    case 13:
+      val = radialStress(tensor, point1, point2, curr_point, direction);
+      break;
+    case 14:
+      val = triaxialityStress(tensor);
+      break;
+    case 15:
+      val = directionValueTensor(tensor, direction);
+      break;
+    default:
+      mooseError("RankTwoScalarAux Error: Pass valid scalar type - " + scalar_type.getRawNames());
```

Comment:
This is not quite the right solution. `scalar_type` is what the user passed in, but it is valid to pass in an _arbitrary_ `MooseEnum` (like one that is built as `"GIRAFFE LION BICYCLE"`). I suggest changing this to `scalarOptions()..getRawNames()`.